### PR TITLE
Replace Vagrant test Setup with Molecule

### PR DESCRIPTION
**Pull request by tnation14**
Replace Vagrant test setup with Molecule, and add support for testing Ubuntu 16.04.
## Testing

Run `molecule test --platform=trusty64` to test Ubuntu 14.04 and `molecule test --platform=xenial64` to test Ubuntu 16.04. Ensure all tests pass.

### CODE REVIEW

File: molecule.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+ansible:
+  config_file: ansible.cfg
```
Comment (hectcastro) [inline molecule.yml:3]: I think the `requirements_file` setting is missing, along with the associated `requirements.yml` file that includes `azavea.java`.

File: playbook.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    - name: Install python
+      raw: apt-get install -yq python
+      become: True
+      when: "{{ ubuntu_release.stdout| version_compare('16.04', '>=') }}"
+
+    - name: Create local facts directory
+      file:
+        path: /etc/ansible/facts.d
+        state: directory
+        mode: u=rw,g=rw,o=rw
```
Comment (hectcastro) [inline playbook.yml]: We don't really have an Ansible playbook style guide, but I think that pretty much all mode definitions we have right now use octal numbers.

File: playbook.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        owner: ubuntu
```
Comment (hectcastro) [inline playbook.yml]: When you run Molecule does it ends up running as `vagrant`?

File: tests/test_scala.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+def test_java_exists(Command):
```
Comment (hectcastro) [inline tests/test_scala.py]: This file has a bunch of tabs in it, which I don't particularly care about, but `flake8` does. I think it would be good to adhere to PEP8 in these test files.

File: tests/test_scala.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+def test_scala_exists(Command, Ansible, SystemInfo):
+	scala_version = Command(
+		"scala -e \"println(util.Properties.versionNumberString)\"")
+	assert scala_version.rc == 0
+
+def test_scala_version(Command, Sudo):
+	with Sudo(user='root'):
+		# Extract target scala version from ansible facts
+		target_scala_version = Command.check_output(
```
Comment (hectcastro) [inline tests/test_scala.py]: Did you try doing anything with the `Ansible()` TI module?

I setup a fixture like this:

``` python
import os.path
import pytest


@pytest.fixture()
def AnsibleDefaults(Ansible):
    return Ansible("include_vars", os.path.join("./defaults/main.yml"))["ansible_facts"]
```

Which provides a data structure like:

``` python
{u'scala_version': u'2.11.1', u'scala_sbt_version': u'0.13.6'}
```

I think this might allow us to test the configured variables from within TI without a template.

File: tests/test_scala.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+			"scala -e \"println(util.Properties.versionNumberString)\" |"
+			" grep %s" % target_scala_version)
+
+		# Extract target sbt version from ansible facts
+		target_sbt_version = Command.check_output(
+			"cat /etc/ansible/facts.d/molecule.fact |"
+			" grep target_sbt_version | cut -f2 -d'='")
+
+		# Ensure that the running version of sbt is the target version
+		Command.run_test("sbt sbtVersion | grep %s" % target_sbt_version)
```
Comment (hectcastro) [inline tests/test_scala.py]: Can we do something like this here instead since SBT is getting installed via package?

``` python
Package("sbt").version == target_sbt_version
```

File: molecule.yml (hunk 1 above)
Comment (tnation14) [inline molecule.yml:3]: Addressed in [`de57c6c`](https://github.com/azavea/ansible-scala/pull/3/commits/de57c6c56ef1134862d8bb76edcb2fbcfc5a15b7)

File: playbook.yml (hunk 3 above)
Comment (tnation14) [inline playbook.yml]: This differs by platform. On Trusty it runs as `vagrant`, in Xenial it runs as `ubuntu`. I ultimately removed the file ownership entirely, though.

File: tests/test_scala.py (hunk 4 above)
Comment (tnation14) [inline tests/test_scala.py]: Addressed in [`99ecbdc`](https://github.com/azavea/ansible-scala/pull/3/commits/99ecbdc317b7d946df54419288b9902daec13154)

File: playbook.yml (hunk 2 above)
Comment (tnation14) [inline playbook.yml]: Fixed in [`9d58d93`](https://github.com/azavea/ansible-scala/pull/3/commits/9d58d93e59721c3a5288cf468e367ceef69ca8e4)

File: tests/test_scala.py (hunk 5 above)
Comment (tnation14) [inline tests/test_scala.py]: I was initially trying to solve the problem of handling version overrides, but that's ultimately not important in these tests. Fixed in [`9d58d93`](https://github.com/azavea/ansible-scala/pull/3/commits/9d58d93e59721c3a5288cf468e367ceef69ca8e4)

File: tests/test_scala.py (hunk 6 above)
Comment (tnation14) [inline tests/test_scala.py]: Fixed in [9d58d93](https://github.com/azavea/ansible-scala/pull/3/commits/9d58d93e59721c3a5288cf468e367ceef69ca8e4)

File: tests/test_scala.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    Command.run_test(
+        "scala -e \"println(util.Properties.versionNumberString)\"")
+
+
+def test_scala_version(Command, AnsibleDefaults):
+    # Extract target scala version from ansible facts
+    target_scala_version = AnsibleDefaults["scala_version"]
+
+    # Ensure that the running version of scala is the target version
+    Command.run_test(
```
Comment (hectcastro) [inline tests/test_scala.py:74]: In the Python community, the `%` method for string formatting has been enhanced with `.format()`. I think that we should switch to using that approach vs. `%` here.

See: https://pyformat.info/

File: tests/test_scala.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    assert Package("openjdk-%s-jdk" % java_major_version).is_installed
+
+
+def test_scala_exists(Command):
+    Command.run_test(
+        "scala -e \"println(util.Properties.versionNumberString)\"")
+
+
+def test_scala_version(Command, AnsibleDefaults):
+    # Extract target scala version from ansible facts
```
Comment (hectcastro) [inline tests/test_scala.py:70]: I think these comments are fine as is, but wanted to point out some of the loose `pydoc` documentation standards from Google we try to follow.

See: https://google.github.io/styleguide/pyguide.html#Comments

File: tests/test_scala.py [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+def test_java_exists(AnsibleFacts, Package):
+
+    # Ubuntu 16.04 and above only support openjdk 8
+    os_version = AnsibleFacts["ansible_distribution_version"]
+    if float(os_version) >= 16.04:
+        java_major_version = "8"
+    else:
+        java_major_version = "7"
+
+    assert Package("openjdk-%s-jdk" % java_major_version).is_installed
```
Comment (hectcastro) [inline tests/test_scala.py]: Same as below regarding string formatting.

File: tests/test_scala.py [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    return Ansible("include_vars", "./defaults/main.yml")["ansible_facts"]
+
+
+@pytest.fixture()
+def AnsibleFacts(Ansible):
+    return Ansible("setup")["ansible_facts"]
+
+
+def test_java_exists(AnsibleFacts, Package):
+
```
Comment (hectcastro) [inline tests/test_scala.py:36]: Unnecessary whitespace here.

**Comment by hectcastro** [comment]
Looks great. Tests are nice and clean. 👍
